src/transform/headers.ts

```diff
@@ -1,15 +1,23 @@
 import { comment } from "../utils";
 import { transformSchemaObj } from "./schema";
 
-export function transformHeaderObjMap(headerMap: Record<string, HeaderObject>): string {
+export function transformHeaderObjMap(
+  headerMap: Record<string, HeaderObject>,
+  { immutableTypes }: { immutableTypes: boolean }
+): string {
   let output = "";
 
   Object.entries(headerMap).forEach(([k, v]) => {
     if (!v.schema) return;
 
     if (v.description) output += comment(v.description);
+
+    const readonly = immutableTypes ? "readonly " : "";
```

Comment:
👍🏻

Reply:
I personally am also a fan of `readonly string[]` over `ReadonlyArray<string>`. AFAIK this looks easier to implement, and has no downsides. I’m in favor of this approach.

Reply:
the "downside" - if you want to call it such - would be a minimum requirement of TypeScript v3.4, which I think should be fine considering v4.2 is currently latest.

Reply:
TS 4 was such an easy update I don’t know of any reason why people would still be on an old version. But given that this is opt-in too, this is more than fine.